engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala

```diff
@@ -230,8 +256,66 @@ object PackageRepository {
         pkg            = pkg,
         isLibrary      = true
       )
+      pkg
     }.toEither.left.map { error => Error.PackageLoadingError(error.getMessage) }
 
+    /** @inheritdoc */
+    override def initialize(): Either[Error, Unit] = {
+      val unprocessedPackages =
+        loadedPackages.keySet
+          .diff(loadedComponents.keySet)
+          .flatMap(loadedPackages(_))
+      unprocessedPackages.foldLeft[Either[Error, Unit]](Right(())) {
+        (result, pkg) =>
+          for {
+            _ <- result
+            _ <- resolveComponentGroups(pkg)
+          } yield ()
+      }
+    }
+
+    private def resolveComponentGroups(
+      pkg: Package[TruffleFile]
+    ): Either[Error, Unit] = {
+      if (loadedComponents.contains(pkg.libraryName)) Right(())
+      else {
+        pkg.config.componentGroups match {
+          case Left(err) =>
+            Left(Error.PackageLoadingError(err.getMessage()))
+          case Right(componentGroups) =>
+            logger.debug(
+              s"Resolving component groups of package [${pkg.name}]."
+            )
+
+            componentGroups.newGroups.foreach(
+              registerComponentGroup(pkg.libraryName, _)
+            )
+            componentGroups.extendedGroups
+              .foldLeft[Either[Error, Unit]](Right(())) {
+                (result, componentGroup) =>
+                  for {
+                    _ <- result
+                    extendedLibraryName = componentGroup.module.libraryName
+                    _ <- ensurePackageIsLoaded(extendedLibraryName)
+                    pkgOpt = loadedPackages(extendedLibraryName)
+                    _ <- pkgOpt.fold[Either[Error, Unit]](Right(()))(
+                      resolveComponentGroups
+                    )
+                  } yield ()
+              }
```

Comment:
If the only purpose of this PR to ensure that libraries referenced in extended component groups are loaded, then the above comment can most likely be ignored.

Reply:
The algorithm is recursive, and `resolveComponentGrous` is called for each extended library as well. All the components should be gathered correctly

Reply:
Yeah, every group is loaded, but the extensions coming to that group from other libraries seem to not be registered anywhere - i.e. seems that `componentGroups.extendedGroups` is only used to find which base groups should be loaded, but the extensions themselves are not registered anywhere. Right?
But maybe at this stage there is no need to register them - just asking because I don't know the exact requirements and found it odd.

Reply:
You're right, it's me being dumb. :facepalm: This code was left before I changed the structure of component groups and I forgot about the extended ones. Fixed